frontend/src/utils/payment-params.ts

```diff
@@ -136,14 +136,15 @@ export class PaymentParams {
   }
 
   private static isDemo(searchParams: URLSearchParams, paramName: string): boolean {
-    if (searchParams.has(paramName)){
+    if (searchParams.has(paramName)) {
       const value = searchParams.get(paramName);
-      if (value?.toLowerCase() !== "true" && value?.toLowerCase() !== "false") {
-        throw new PaymentParamError("demo should be a boolean");
+      if (value?.toLowerCase() === "true") {
+        return true;
+      } else {
+        return false;
       }
-      return true;
     } else {
-      return false
+      return false;
     }
```

Comment:
We can simplify this section:
`
    if (searchParams.has(paramName)) {
      const value = searchParams.get(paramName);
      if (value?.toLowerCase() === "true") {
        return true;
      }
      return false;
    }
`

Reply:
The code you provided results into a TS error because isDemo is supposed to be returning a boolean. If this condition is not true:
if (searchParams.has(paramName)) nothing is returned. 

I adjusted it like so:

` if (searchParams.has(paramName)) {
      const value = searchParams.get(paramName);
      if (value?.toLowerCase() === "true") {
        return true;
      }
      return false;
    } else {
      return false;
    }`